src/transactions/TransactionFrame.cpp

```diff
@@ -1351,9 +1199,11 @@ TransactionFrame::commonValid(Application& app,
                               bool chargeFee,
                               uint64_t lowerBoundCloseTimeOffset,
                               uint64_t upperBoundCloseTimeOffset,
-                              std::optional<FeePair> sorobanResourceFee)
+                              std::optional<FeePair> sorobanResourceFee,
+                              TransactionResultPayloadPtr resPayload) const
 {
     ZoneScoped;
+    releaseAssertOrThrow(resPayload);
```

Comment:
Can we just accept a reference instead?

Reply:
No, same issue as above, this transitively calls `checkSorobanResourceAndSetError`.